### PR TITLE
Merge ACK and ACK_ECN

### LINKED ISSUE #1778: Merge ACK and ACK_ECN
Discussed in NYC, we agreed to keep the current design, but make an editorial change so that there is only one name.

### PR DESCRIPTION
**Pull request by ianswett**
Fixes #1778 

Also changes the type value for ACK, ACK_ECN, and RETIRE_CONNECTION_ID in order to give the ACK frame two adjacent values.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -3444,15 +3447,19 @@ ACK Blocks:
...
+variable number of alternating Gap and Additional ACK Blocks.  The number of
+Gap and Additional ACK Block fields is determined by the ACK Block Count field.
 
 Gap and ACK Block fields use a relative integer encoding for efficiency.  Though
 each encoded value is positive, the values are subtracted, so that each ACK
 Block describes progressively lower-numbered packets.  As long as contiguous
 ranges of packets are small, the variable-length integer encoding ensures that
 each range can be expressed in a small number of octets.
 
+The ACK frame uses the least significant bit(type=0x1b) to indicate ECN feedback
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]: "bit (that is, type 0x1b)"

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -3374,9 +3373,11 @@ Application Error Code:
 
 ## ACK Frame {#frame-ack}
 
-Receivers send ACK frames (type=0x0d) to inform senders which packets they have
-received and processed. The ACK frame contains any number of ACK blocks.
-ACK blocks are ranges of acknowledged packets.
+Receivers send ACK frames (types=0x1a - 0x1b) to inform senders of packets they
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]: (types 0x1a and 0x1b)

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -3374,9 +3373,11 @@ Application Error Code:
...
+have received and processed. The ACK frame contains one or more ACK blocks. ACK
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]: should we be capitalizing the B in ACK Blocks?

**Comment by ianswett** [comment]
I decided against the big renumber.  I thought it'd be best done interactively by 2 or 3 people together.

Block wasn't consistently capitalized, so I made it so everywhere.

**Comment by ianswett** [comment]
Any other comments?  Martin or Jana, would you be willing to merge this?

**Pull request merged**